Add Search and Account to Header

tuchfarber opened the pull request.

bill-filler commented:
couple of functional issues/questions:

1. header seems to be incorrectly disappearing when pressing the Content menu and then left with 2 Content buttons (see attachment) 

![screenshot from 2018-08-27 12-22-19](https://user-images.githubusercontent.com/27503539/44671812-87303d00-a9f4-11e8-9ee7-399d50e1b798.png)

1. Do you have the correct breakpoints set for moving to mobile header? Doesn't switch to mobile header when I "Inspect" and choose any of the Iphone modes and I expected it to. Does switch if I manually resize the browser past a certain point.

1. In mobile layout, after pressing the Search button you should move the focus to the search field

bill-filler commented:
Also missing "My Profile" menu option in "Account" dropdown which is supposed to take you to account settings in LMS https://courses.edx.org/account/settings
See https://app.zeplin.io/project/5b21820665ca1be0451f05a9/screen/5b27e5e54540d51cb74de072

bill-filler commented:
Mobile header supposed to be "sticky" on scroll:
https://app.zeplin.io/project/5b21820665ca1be0451f05a9/screen/5b33f760986fa3ea5df60260
Seems like a nice to have - if this is hard/time consuming just file a ticket so we can track, otherwise implement if not difficult.

bill-filler commented:
In mobile mode, clicking on a link in the TOC does not display the page

bill-filler commented:
In mobile mode, TOC not supposed to obscure page beneath it fully:
https://app.zeplin.io/project/5b21820665ca1be0451f05a9/screen/5b27e5e4104ed15d16bd350c
Seems like a nice to have - if this is hard/time consuming just file a ticket so we can track, otherwise implement if not difficult.

bill-filler commented:
In mobile mode Account button menu supposed to take full-width:
https://app.zeplin.io/project/5b21820665ca1be0451f05a9/screen/5b27e5e4cb5696cb1200ae13
Seems like a nice to have - if this is hard/time consuming just file a ticket so we can track, otherwise implement if not difficult.

tuchfarber commented:
1. Header incorrectly disappearing when content menu pressed
    - Fixed in d5e0f50dad9425c1589c0773943787cd075ca064
2. Doesn't switch to mobile on 'inspect' iPhone in chrome
    - Fixed in d843db1f7365a0ec3d03f5083cae56a4867a3ef5
3. Move focus to search field after hitting search button
    - Blocked by: https://github.com/edx/paragon/pull/322
    - WL-1775
4. Add My Profile button to Account
    - Fixed in d23b508ee575c206be366a856e85cf3a6c978de5
5. Make mobile header sticky
    - Fixed in 89c39d6f2efd7b5049ae85e70009206958e85ef7
6. Mobile Mode doesn't change page on TOC click
    - Fixed in d486e5cf24d1a856814914cd14807a76ee4643ed
7. Mobile mode shouldn't obscure full page
    - Requires further refactoring / 3 points
    - WL-1773
8. Account dropdown should be full width in mobile mode
    - Requires further refactoring / 2 points
    - WL-1774

bill-filler commented on src/components/SiteHeader/index.jsx:
```diff
@@ -0,0 +1,80 @@
+              menuItems={
+                [
+                  { label: 'Logout', href: props.logoutPath },
+                  { label: 'My Account', href: props.lmsAccountPath },
```
Couple of minor things to match the design:
- Logout option should be last
- "My Account" should be "My Profile" per design
- "My Account" should have an icon
